osf/models/mixins.py

```diff
@@ -263,7 +263,7 @@ def add_addon(self, addon_name, auth=None, override=False, _force=False):
         model = self._settings_model(addon_name, config=config)
         ret = model(owner=self)
         ret.on_add()
-        ret.save()  # TODO This doesn't feel right
+        ret.save(clean=False)  # TODO This doesn't feel right
```

Comment:
I'm fine with this, but can you explain why we don't need the full-clean here?

Reply:
Full clean checks to make sure that all the foreign keys actually exist so it can give a nice error message. This method creates all the rows that would be checked so it's a moot point.
I think I added this in before disabling it for all tests, so this could be removed if you'd prefer that.

Reply:
Ah, thanks for the clarification. This is fine to keep.